python/tests/integration/arcticdb/version_store/test_snapshot.py

```diff
@@ -491,4 +498,343 @@ def assert_0_delete_marker(lib, storage):
+
+@pytest.mark.xfail(reason = "ArcticDB#1863")    
+def test_delete_snapshot_complex_flow_with_delete_multible_smbls(basic_store_tiny_segment_dynamic):
+    lib = basic_store_tiny_segment_dynamic
+
+    symbol1 = "sym1"
+    df_1_0 = create_df_index_rownum(10, 0, 10)
+    df_1_1 = create_df_index_rownum(10, 10, 200)
+    df_1_2 = create_df_index_rownum(10, 1000, 1300)
+    df_1_combined_0_1 = pd.concat([df_1_0, df_1_1])
+    df_1_combined_0_2 = pd.concat([df_1_0, df_1_2])
+    df_1_combined = pd.concat([df_1_0, df_1_1, df_1_2])
+    lib.write(symbol1, df_1_0)
+    lib.append(symbol1, df_1_1)
+    lib.append(symbol1, df_1_2)
+
+    symbol2 = "sym2"
+    df_2_0 = create_df_index_rownum(10, -100, -90)
+    df_2_1 = create_df_index_rownum(10, -80, -50)
+    df_2_combined = pd.concat([df_2_0, df_2_1])
+    lib.write(symbol2, df_2_0)
+    lib.append(symbol2, df_2_1)
+
+    symbol3 = "sym3"
+    df_3_0 = create_df_index_rownum(10, 1000, 1010)
+    df_3_1 = create_df_index_rownum(10, 2000, 2210)
+    df_3_combined = pd.concat([df_3_0, df_3_1])
+    lib.write(symbol3, df_3_0)
+    lib.append(symbol3, df_3_1)
+
+    snap1 = "snap1"
+    snap2 = "snap2"
+    snap1_vers = {symbol1 : 1, symbol2 : 1}
+    lib.snapshot(snap1, versions=snap1_vers)
+
+    # verify initial state
+    assert sorted(lib.list_snapshots()) == [snap1] 
+    assert_frame_equal(df_1_combined, lib.read(symbol1).data)      
+    assert_frame_equal(df_2_combined, lib.read(symbol2).data)      
+    assert_frame_equal(df_3_combined, lib.read(symbol3).data)      
+
+    lib.delete_version(symbol1, 1)
+    lib.delete_version(symbol2, 1)
+
+    # confirm afer deletion of versions all is as expected
+    assert sorted(lib.list_snapshots()) == [snap1] 
+    assert df_1_combined.equals(lib.read(symbol1).data)
+    assert df_2_0.equals(lib.read(symbol2).data)
+    assert df_3_combined.equals(lib.read(symbol3).data)
+    # verify sumbol 1
+    assert len(lib.list_versions(symbol1)) == 3
+    assert [ver["deleted"] for ver in lib.list_versions(symbol1)] == [False, True, False]
+    assert_frame_equal(df_1_combined_0_1, lib.read(symbol1, as_of=snap1).data)      
+    assert_frame_equal(df_1_combined_0_1, lib.read(symbol1, as_of=snap1_vers[symbol1]).data)      
+    # verify sumbol 2
+    assert len(lib.list_versions(symbol2)) == 2
+    assert [ver["deleted"] for ver in lib.list_versions(symbol2)] == [True, False]
+    assert_frame_equal(df_2_combined, lib.read(symbol2, as_of=snap1).data)      
+    assert_frame_equal(df_2_combined, lib.read(symbol2, as_of=snap1_vers[symbol2]).data)      
+    # verify sumbol 3
+    assert [ver["deleted"] for ver in lib.list_versions(symbol3)] == [False, False]
+
+    lib.delete_snapshot(snap1)
+
+    # confirm afer deletion of versions all is as expected
+    # as well as deleting the snapshot wipes the versions effectivly
+    assert sorted(lib.list_snapshots()) == [] 
+    assert_frame_equal(df_1_combined_0_2, lib.read(symbol1).data)      
```

Comment:
This assumption is wrong. Deleting version 1 will not delete the data for for version one, because the data is referenced by version 2.

Reply:
indeed - now fixed to:
assert_frame_equal(df_1_combined, lib.read(symbol1).data)